var/spack/repos/builtin/packages/cray-mvapich2/package.py

```diff
@@ -29,29 +30,10 @@ class CrayMvapich2(Package):
     requires("platform=linux", msg="Cray MVAPICH2 is only available on Cray")
 
     def setup_run_environment(self, env):
-        if spack_cc is None:
-            return
-
-        env.set("MPICC", spack_cc)
-        env.set("MPICXX", spack_cxx)
-        env.set("MPIF77", spack_fc)
-        env.set("MPIF90", spack_fc)
-
-    def setup_dependent_build_environment(self, env, dependent_spec):
-        dependent_module = dependent_spec.package.module
-        env.set("MPICH_CC", dependent_module.spack_cc)
-        env.set("MPICH_CXX", dependent_module.spack_cxx)
-        env.set("MPICH_F77", dependent_module.spack_f77)
-        env.set("MPICH_F90", dependent_module.spack_fc)
-        env.set("MPICH_FC", dependent_module.spack_fc)
-
-    def setup_dependent_package(self, module, dependent_spec):
-        spec = self.spec
-        dependent_module = dependent_spec.package.module
-        spec.mpicc = dependent_module.spack_cc
```

Comment:
This one was added in #26273 and the recipe has never been updated since. `cray-mpich` was added before, in #20076, and `cray-mvapich2` seemed to be taken from it. `cray-mpich` was then updated in #25597 and #30249, so I assumed we should do the same also here. 

It's true that now there is this regression from Cray in newer PE, so :shrug:

Reply:
Hmm, yeah, I think simplicity here is best. I will work on finding a machine to test this on, but I think having all of these behave the same and avoid drift is probably a good idea.